src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -424,6 +434,14 @@ internal void MarkThreadRequestSatisfied()
             // by the time we reach this point.
             //
             int count = numOutstandingThreadRequests;
+            if (dequeSuccessful && ThreadRequestNeeded(count))
```

Comment:
Why does it need to request a thread when request count < proc count? For instance suppose proc count is 8, initially 4 work items were enqueued, the work item count at the beginning of this function is 3 (this thread just finished dequeuing a work item), and the request count is 4. `ThreadRequestNeeded(count - 1)` would return true when using `workItems.IsEmpty()` and this function would request a thread unnecessarily.